### ISSUE TITLE
HTTP clients

### ISSUE THREAD

**Issue by elmigranto** [j3k0/ganomede#14]
I've been a bit stubborn in using `request` instead of `restify`'s clients, but finally got around to figuring it out in mailchimp worker. Pretty neat, espesially considering how awkward, bloaty and buggy `request` is (at least for our use-case). Would be nice to move everything to `restify.JsonClient`, better yet, some light wrapper on top of it, so we can do following things:

### `X-Request-Id`

As you mentioned, crucial thing with microservices.

### Retries

When implementing mailchimp worker, found following passage:

> All clients support retry with exponential backoff for getting a TCP connection; they do not perform retries on 5xx error codes like previous versions of the restify client.

Initially, I disabled it thinking 5xx is a retry, but looks like it's safe to enable. Maybe we even want to rety on some of `RestError`s, though, I can't quite think about any when retry would solve an issue.

<hr/>

As I'm writing this down, I realize this is kind of low on actionable info… 😕

**Comment by j3k0** [comment]
A small check-list of places where request is used in place of JsonClient. Here's what a quick `grep -r` on all ganomede services found (excepting tests and other places where it doesn't matter).

 - [x] ./ganomede-events/src/client/Client.js
 - [x] ./ganomede-events/src/client/request.js
 - [ ] ./ganomede-wordsaxe/src/words/fetch-from-data.js

If I didn't miss anything, it means it should be quick to fix.

**Comment by elmigranto** [comment]
It's also in `ganomede-admin`.

**Comment by j3k0** [comment]
Yes, `ganomede-admin` is one of those places where is doesn't really matter.
